### DISCUSSION TITLE
Light intensity confusion

### DISCUSSION THREAD

**Discussion by jwwalker** [google/filament#3866] [Q&A]
I notice that if you set the intensity of a spot light in lumens, and then call `getIntensity`, you get a different number back, apparently the original number divided by pi?  Is that a bug, or some subtlety I'm not aware of?

**Comment by romainguy** [comment]
It's a question of luminous intensity vs power but the documentation doesn't match the behavior. Either should document what we actually return or return the original value.

@bejado Could you fix this please?

**Reply by bejado** [reply]
Where is the documentation wrong?

For `Type::SPOT`, `LightManager.setIntensity` sets the luminous _power_ of the light. `LightManager.getIntensity` returns the luminous _intensity_. [Equation 61](https://google.github.io/filament/Filament.html#mjx-eqn-spotLightLuminousPowerB) explains the relationship and the pi term.

**Reply by bejado** [reply]
Oh I see, `getIntensity` has _lumen_ listed as the unit of intensity, which should be _candela_.

**Reply by romainguy** [reply]
We may want to add more details to the docs: setIntensity does mention the luminous power in lumen, and getIntensity the luminous intensity in lumen. It might be helpful if we defined those two terms.
